blog/fine-grained-news-2025-10.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Fine Grained News - October 2025
+description: Fine Grained News
+slug: fine-grained-news-2025-10
+date: 2025-10-15
+authors: aaguiar
+tags: [newsletter]
+image: https://openfga.dev/img/og-rich-embed.png
+hide_table_of_contents: false
+---
+# Fine Grained News - October 2025
+
+Welcome to OpenFGA's Fine Grained News, October edition!
+
+## 🎉 OpenFGA to CNCF Incubation!
+
+The CNCF completed the Due Diligence process to approve moving OpenFGA to the CNCF Incubation stage, and it's now [open for comments](https://github.com/cncf/toc/pull/1923
+). Support emojis are appreciated! :)
+
+Thanks to Grafana, Docker, Read.AI, Agicap, and Zuplo for taking the time to talk about their experience to the CNCF! 
+
+Moving to Incubation is great news for the OpenFGA community and we are super excited about it!
+
+## 🚀 What We've Shipped
+
+### OpenFGA v1.10.0 Performance Improvements
+
+Starting with [OpenFGA v1.10.0](https://github.com/openfga/openfga/releases/tag/v1.10.0), we've included a query planner that selects resolution strategies based on runtime statistics, behind the `enable-check-optimization` flag. The planner can pick different strategies per-relation, and we've seen significant improvements in performance for some models/relations.
+
+### Write Endpoint Enhancements  
+
+In case you missed it, it also adds additional parameters to the [`/write`](https://openfga.dev/api/service#/Relationship%20Tuples/Write) endpoint that allow specifying the behavior when duplicated tuples are written or non-existing tuples are deleted. We'll be completing SDK support for this feature in the next few weeks. 
+
+### SDK & Tooling Updates
+
+- [Go SDK 0.7.3](https://github.com/openfga/go-sdk/releases/tag/v0.7.3) adds support for specifying [custom headers per request](https://github.com/openfga/go-sdk#custom-headers), and to specify [conflict resolution options for Write operations](https://github.com/openfga/go-sdk/blob/v0.7.3/README.md#conflict-options-for-write-operations).
+
+- [.NET SDK 0.7.0](https://github.com/openfga/dotnet-sdk/releases/tag/v0.7.0) adds support for .NET Standard 2.0, .NET 8 and .NET 9. It builds on the [0.6.0 version](https://github.com/openfga/dotnet-sdk/releases/tag/v0.6.0) released a few days earlier, which has support for specifying contextual tuples and context in Assertions, contextual tuples in Expand, and other improvements.
+
+- [Java SDK 0.9.1](https://github.com/openfga/java-sdk/releases/tag/v0.9.1) fixes issues when overriding request headers and correctly sends headers in the listRelations method.
```

Comment:
```suggestion
- [Java SDK 0.9.1](https://github.com/openfga/java-sdk/releases/tag/v0.9.1) fixes issues when overriding request headers.
```

Just the other part is not important enough for end users to get called out here - it's still in the release docs.